### PR TITLE
ENH: to_csv and read_{fwf/csv/table} type improvements

### PR DESCRIPTION
**Pull request by bashtage**
- [X] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Comment by bashtage** [comment]
@Dr-Irv Looks like the loguru stuff fixed itself?

**Comment by bashtage** [comment]
Maybe not fixed, but random.  Might depend on the CI worker node.

**Comment by Dr-Irv** [comment]
> Maybe not fixed, but random. Might depend on the CI worker node.

Yes, this randomness is part of the problem.  I don't know what to do about it.

File: pandas-stubs/core/generic.pyi [hunk 1]
```diff
@@ -204,17 +205,19 @@ class NDFrame(PandasObject, indexing.IndexingMixin):
...
-        columns: Sequence[Hashable] | None = ...,
+        float_format: _str | Callable | None = ...,
+        columns: list[HashableT] | None = ...,
         header: _bool | list[_str] = ...,
         index: _bool = ...,
-        index_label: _bool | _str | Sequence[Hashable] | None = ...,
-        mode: _str = ...,
+        index_label: Literal[False] | _str | list[HashableT] | None = ...,
+        mode: Literal[
+            "a", "w", "x", "at", "wt", "xt", "ab", "wb", "xb", "w+", "w+b", "a+", "a+b"
```
Comment (Dr-Irv) [inline pandas-stubs/core/generic.pyi:217]: Maybe create a type for this?

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
My only concern here is that I can't tell if you modified `read_csv()` or `read_table()` because they moved from one file to the other.

If you did change `read_csv()` or `read_table()`, can you do one PR that just modifies their signatures, but keeping them in the same file, so it is easy for me to compare what you changed, and then a second PR that moves them from where they are in `main` now to the new place you put them.

File: pandas-stubs/core/generic.pyi (hunk 1 above)
Comment (bashtage) [reply pandas-stubs/core/generic.pyi:217]: Fixed in other PR.

**Comment by bashtage** [comment]
> If you did change `read_csv()` or `read_table()`, can you do one PR that just modifies their signatures, but keeping them in the same file, so it is easy for me to compare what you changed, and then a second PR that moves them from where they are in `main` now to the new place you put them.

Created #248 in place of this

**Pull request closed without merging**